components/BookmarkButton.js

```diff
@@ -1,17 +1,18 @@
 import styled from "styled-components";
-import BookmarkPinIcon from "assets/bookmarkicons/BookmarkPinIcon.svg";
+import BookmarkIcon from "assets/bookmarkicons/Bookmark.svg";
+import BookmarkIconFilled from "assets/bookmarkicons/Bookmark-filled.svg";
 
 export default function BookmarkButton({ isBookmarked, onToggleBookmark }) {
   return (
     <StyledButton onClick={onToggleBookmark} aria-label="Toggle Bookmark">
-      <StyledBookmarkPinIcon $isBookmarked={isBookmarked} />
+      {isBookmarked ? <StyledBookmarkIconFilled /> : <StyledBookmarkIcon />}
```

Comment:
I guess you don’t necessarily need two separate styled components (`StyledBookmarkIcon` and `StyledBookmarkIconFilled`). You could conditionally render the icons directly in the JSX, simplifying your code.

```suggestion
      {isBookmarked ? <BookmarkIconFilled /> : <BookmarkIcon />}
```